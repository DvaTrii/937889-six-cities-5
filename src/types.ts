export type OfferCardType = {
  id: number,
  city: {
    name: string,
    coordinates: {
      latitude: number,
      longitude: number,
      zoom: number
    }
  },
  coordinates: {
    latitude: number,
    longitude: number,
    zoom: number
  },
  pictures: Array<string>,
  previewImage: string,
  isPremium?: boolean,
  isBookmark?: boolean,
  price: number,
  title: string,
  type: string,
  rating: number,
  description: string,
  bedroomsMax: number,
  guestsMax: number,
  amenities: Array<string> | null,
  hostInfo: {
    avatar: string,
    name: string,
    isSuper?: boolean,
  }
};

export enum Cities {
  PARIS = `Paris`,
  COLOGNE = `Cologne`,
  BRUSSELS = `Brussels`,
  AMSTERDAM = `Amsterdam`,
  HAMBURG = `Hamburg`,
  DUSSELDORF = `Dusseldorf`,
}

export type ReviewType = {
  id: number,
  date: number,
  text: string,
  rating: number,
  userInfo: {
    avatar: string,
    name: string,
    id: number,
    isSuper: boolean,
  }
};
