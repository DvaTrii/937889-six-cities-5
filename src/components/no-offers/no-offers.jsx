import React from "react";
import PropTypes from "prop-types";

const NoOffers = ({city}) => {
  return (
    <div className="cities">
      <div className="cities__places-container cities__places-container--empty container">
        <section className="cities__no-places">
          <div className="cities__status-wrapper tabs__content">
            <b className="cities__status">No places to stay available</b>
            <p className="cities__status-description">We could not find any property available at the moment in {city}</p>
          </div>
        </section>
        <div className="cities__right-section"></div>
      </div>
    </div>
  );
};

NoOffers.propTypes = {
  city: PropTypes.string.isRequired,
};

export default NoOffers;
