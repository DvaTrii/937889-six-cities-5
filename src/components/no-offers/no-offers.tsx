import * as React from "react";
import {connect} from "react-redux";

import {getActiveCity} from "../../store/app/selectors";

type Props = {
  city: string
};

const NoOffers: React.FC<Props> = ({city}) => {
  return (
    <div className="cities">
      <div className="cities__places-container cities__places-container--empty container">
        <section className="cities__no-places">
          <div className="cities__status-wrapper tabs__content">
            <b className="cities__status">No places to stay available</b>
            <p className="cities__status-description">We could not find any property available at the moment in {city}</p>
          </div>
        </section>
        <div className="cities__right-section"/>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return ({
    city: getActiveCity(state)
  });
};

export {NoOffers};
export default connect(mapStateToProps)(NoOffers);
