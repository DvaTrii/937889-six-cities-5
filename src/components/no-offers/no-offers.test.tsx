import * as React from "react";
import * as renderer from "react-test-renderer";
import {NoOffers} from "./no-offers";
import {Cities} from "../../types";

it(`Should NoOffers render properly`, () => {
  const city: Cities = Cities.AMSTERDAM;

  const tree = renderer
    .create(
        <NoOffers
          city={city}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
