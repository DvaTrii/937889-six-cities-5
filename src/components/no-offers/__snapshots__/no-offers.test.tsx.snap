// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should NoOffers render properly 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container cities__places-container--empty container"
  >
    <section
      className="cities__no-places"
    >
      <div
        className="cities__status-wrapper tabs__content"
      >
        <b
          className="cities__status"
        >
          No places to stay available
        </b>
        <p
          className="cities__status-description"
        >
          We could not find any property available at the moment in 
          Amsterdam
        </p>
      </div>
    </section>
    <div
      className="cities__right-section"
    />
  </div>
</div>
`;
